src/vdaf.rs

```diff
@@ -374,10 +374,22 @@ pub fn verify_start<V: Value>(
 /// [`VerifierMessage`] messages broadcast by all of the aggregators and produces the aggregator's
 /// input share.
 // TODO(cjpatton) Check for ciphersuite mismatch between `state` and `msgs` and among `msgs`.
-pub fn verify_finish<V: Value>(
+pub fn verify_finish<M, E, V, const N: usize>(
     state: AggregatorState<V>,
-    msgs: Vec<VerifierMessage<V::Field>>,
-) -> Result<V, VdafError> {
+    msgs: M,
+) -> Result<V, VdafError>
+where
+    V: Value,
+    M: TryInto<[VerifierMessage<V::Field>; N], Error = E>,
+    E: std::fmt::Debug,
```

Comment:
It would be nice to reduce the number of type parameters. Is this possible?
```suggestion
    M: TryInto<[VerifierMessage<V::Field>; N], Error: std::fmt::Debug>,
```
It would avoid the additional type parameter `E`. (Also, I think `Debug` is imported by default these days.)

Alternatively, I would be comfortable with a `panic!()` here instead of propagating the error. This is less of an input validation issue than it is a programmer error.

Reply:
I tried doing that but you can't do it yet. There's a [Rust RFC](https://rust-lang.github.io/rfcs/2289-associated-type-bounds.html), [it's implemented](https://github.com/rust-lang/rust/pull/57428), but it's [not yet stabilized](https://github.com/rust-lang/rust/issues/52662).

Reply:
In that case my preference would be to panic instead of propagating the error.

Reply:
Sure, I struggle to imagine this failing for any reason beyond whatever collection `msg` is having the wrong number of elements, so I think we can just `panic!` and get rid of the `Debug` trait bound.

Reply:
cjpatton and I chatted about this and got extremely galaxy brained about it. I'm going to try changing this so that `verify_finish` takes `IntoIterator<Item = VerifierMessage<V::Field>>`, which ✅ allows callers to provide any kind of slice or a vec or whatever and ✅ is infallible so we don't have to deal with any type bounds on errors.